tests/h/services/profile_group_test.py

```diff
@@ -122,3 +204,23 @@ def authority_group_service(pyramid_config):
     service.public_groups.return_value = None
     pyramid_config.register_service(service, name='authority_group')
     return service
+
+
+@pytest.fixture
+def pyramid_request(pyramid_request):
+    pyramid_request.route_url = mock.Mock(return_value='/group/a')
+    return pyramid_request
+
+
+@pytest.fixture
+def profile_group_factory(db_session, pyramid_request):
+    """ Returns a callable that will create a ProfileGroupService
+        with a fake AuthorityGroupService whose `public_groups` will return the
+        passed `open_groups`"""
```

Comment:
Fwiw I think strict [pep257](https://www.python.org/dev/peps/pep-0257/) style for this docstring would be something like:

```python
"""
Return a callable that will create a ProfileGroupService.

Return a callable that will create a ProfileGroupService with a fake AuthorityGroupService
whose ``public_groups`` will return the passed ``open_groups``.

"""
```

Note _return_ not _returns_ (imperative tense, tends to make things slightly shorter and more readable) and double instead of single backticks (using [reStructuredText](http://www.sphinx-doc.org/en/stable/rest.html#) instead of markdown).

rST instead of markdown this is because we _could_ one day use [Sphinx autodoc](http://www.sphinx-doc.org/en/stable/ext/autodoc.html) to extract API docs from our code into <http://h.readthedocs.io/> although we don't currently. This may not seem that useful - who needs to browse method signatures and docstrings by HTML, when you can just read the source code? But it does have its benefits as Sphinx has ways for a docstring to reference an object or method or type elsewhere in the code and will hyperlink the reference and will also _throw warnings if these references go out of date_. And a handful of other handy features like that. So it can actually help us to keep our docstrings up to date. So _maybe_ we'll actually use it one day!

You can also use Sphinx [info field lists](http://www.sphinx-doc.org/en/stable/domains.html#info-field-lists) when you want to document params, returns and exceptions (_not_ required for every docstring!)

Reply:
This is really helpful and I will update accordingly. It's these little things that are gonna trip me up on my re-immersion to python!